### PR TITLE
New Feature Grid With Custom Backgrounds

### PR DESCRIPTION
**Pull request by echen-adobe**
Describe your specific features or fixes

Implements an update to the feature-grid block with properties that are more flexible.

Users can now specify new properties within the block as pictured below.
 
This allows the user to customize the background of the feature grid cards and the load more button more effectively. 

Resolves: [MWPW-153691](https://jira.corp.adobe.com/browse/MWPW-153691)

Test URLs:
- Before: https://main--express--adobecom.hlx.page/express/
- After:  https://feature-grid--express--adobecom.hlx.page/drafts/echen/feature-grid-desktop
 
Screenshot 2024-07-17 at 2 32 07 PM 
Screenshot 2024-07-09 at 2 43 38 PM
Screenshot 2024-07-09 at 2 43 43 PM

### CODE REVIEW

**Review by JingleH — APPROVED** [review]

**Review by kennethlum — APPROVED** [review]

File: express/blocks/feature-grid/feature-grid.js [hunk 1]
```diff
@@ -71,11 +77,36 @@ function getLoadMoreText(rows) {
...
+const extractProperties = (block) => {
+  const rows = Array.from(block.querySelectorAll(':scope > div'));
+
+  const allProperties = {};
+
+  rows.forEach((row) => {
+    if (row?.children?.length !== 2) {
+      return;
+    }
+    console.log(row);
```
Comment (fullcolorcoder) [inline express/blocks/feature-grid/feature-grid.js:91]: Stray console

File: express/blocks/feature-grid/feature-grid.js [hunk 2]
```diff
@@ -71,11 +77,36 @@ function getLoadMoreText(rows) {
...
+    const key = row?.children[0].textContent;
+    const value = row?.children[1].textContent;
+    if (key && value && blockProperties.includes(key?.toLowerCase())) {
+      allProperties[key.toLowerCase()] = value;
```
Comment (fullcolorcoder) [inline express/blocks/feature-grid/feature-grid.js:95]: No test coverage

File: express/blocks/feature-grid/feature-grid.js [hunk 3]
```diff
@@ -71,11 +77,36 @@ function getLoadMoreText(rows) {
...
+};
+
 export default function decorate(block) {
+  const colorProperties = extractProperties(block);
   const inputRows = block.querySelectorAll(':scope > div > div');
   block.innerHTML = '';
   const rows = Array.from(inputRows);
   const heading = rows.shift();
+
+  console.log(colorProperties);
```
Comment (fullcolorcoder) [inline express/blocks/feature-grid/feature-grid.js:109]: Stray console

**Review by fullcolorcoder — CHANGES_REQUESTED** [review]

**Review by fullcolorcoder — APPROVED** [review]
Not standing in the way of merging, but it'd be awesome if you could clean up the linting errors so they don't pile up.

image

**Pull request merged**